osu.Desktop/OsuGameDesktop.cs

```diff
@@ -117,6 +117,19 @@ private void fileDrop(object sender, FileDropEventArgs e)
         /// </summary>
         private class StableStorage : WindowsStorage
         {
+            private const string default_songs_path = "Songs";
+
+            private string songsPath;
+            private bool usesCustomSongsPath;
+            private DesktopGameHost host;
+
+            public StableStorage(DesktopGameHost host)
+                : base(string.Empty, host)
+            {
+                this.host = host;
+                songsPath = locateSongsDirectory();
+            }
+
             protected override string LocateBasePath()
             {
                 static bool checkExists(string p) => Directory.Exists(Path.Combine(p, "Songs"));
```

Comment:
I know this is not your code, but this check seems very fragile - it might fail if someone deletes/renames the original `Songs` folder, which isn't that unlikely to happen if we're talking about changing the beatmap directory

imo we could just check for the `osu!.exe` there instead.